### DISCUSSION TITLE
VST3 program 1 in Reaper preset dropdown menu?

### DISCUSSION THREAD

**Discussion by ghjameslo** [pierreguillot/Camomile#289] [Q&A] [answered]
I've created several Camomile VST3 plug-ins that do not declare any programs in the associated text file, yet in the Reaper preset dropdown menu, there is always an entry for "Program 1" under "-- VST built-in programs -".  What am I misunderstanding?

PS Thanks for writing Camomile, it makes me very happy!

**Comment by pierreguillot** [comment, marked as answer]
Hello,
The minimum number of programs is 1. That is defined by the plugin specification (see this [line](https://github.com/juce-framework/JUCE/blob/e6ec1819ec0a59a7cfe82d8ce72367f64e0a4bb0/modules/juce_audio_processors/processors/juce_AudioProcessor.h#L1045) in Juce). So even if you don't specify any program, there is one by default.

**Reply by ghjameslo** [reply]
I wonder if it has more to do with Juce than the VST3 spec because I have several commercial VST3 plug-ins (Melodyne, SpectralLayers, and RX7) that don't have programs in their preset dropdown menus?

**Reply by pierreguillot** [reply]
Yes, it might be possible but I can't bypass Juce... I'll look if there is any thread about it on https://forum.juce.com/.

**Reply by ghjameslo** [reply]
It's not a big deal, I was just wondering if I had made a mistake.  Everything else is just curiosity :)
